### ISSUE TITLE
[azure-eventgrid] REvisit samples

### ISSUE THREAD

**Issue by t-swpill** [Azure/azure-sdk-for-python#12933]
Logging issue for future reference. For all samples in track 2 EventGrid preview, docstring needs to be added to the samples similar to: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/samples/blob_samples_authentication.py#L10-L25. 
The samples are currently available here: https://github.com/t-swpill/azure-sdk-for-python/tree/event-grid-track-2/sdk/eventgrid/azure-eventgrid/samples
Once the PR is merged, they will be available at `azure\azure-sdk-for-python\sdk\eventgrid\azure-eventgrid\samples`.

Known ISsues:

- Add docstrings
- remove pluggables
- revisit "SCRIPT_DIR "
- revisit the gitignore file in samples folder

Edit by Kibrantn on 8/28: Need to also add post-deserialize-use-cases to samples to show diving into and consuming event payload.

**Comment by rakshith91** [comment]
Added docstrings - closing this in favor of https://github.com/Azure/azure-sdk-for-python/issues/13582
